### PR TITLE
feat: Kotlin 2.1.0-Beta2 EAP release details

### PR DESCRIPTION
**Pull request by koshachy**

### CODE REVIEW

File: docs/topics/whatsnew-eap.md [hunk 1]
```diff
@@ -67,19 +67,110 @@ the Kotlin team at JetBrains is focusing on improving the language with new feat
...
+    kvs.store("") { 1 } // Resolves to 2
+
+    // Extension functions
+    kvs.storeExtension("", 1)    // Resolves to 1
+    kvs.storeExtension("") { 1 } // Doesn't resolve
+}
+```
+
+In the example, the `KeyValueStore` class has two overloads for the `store()` function,
+where one overload has function parameters with generic types `K`, `V`,
```
Comment (AlejandraPedroza) [inline docs/topics/whatsnew-eap.md:159]: Maybe `<K, V>`  OR " `K` and `V`, "

File: docs/topics/whatsnew-eap.md [hunk 2]
```diff
@@ -120,13 +211,185 @@ before the new implementation is stabilized.
...
+
+This behavior differed from JavaScript and TypeScript,
+which allow such properties to be accessed using non-identifier characters.
+
+Starting from Kotlin %kotlinEapVersion%, this feature is enabled by default.
+Kotlin/JS now allows you to use the backquote syntax and the `@JsName` annotation
+to interact with JavaScript properties that contain non-identifier characters and to use names for test methods.
+
+Now, you can enclose property names in backticks (``) to reference non-identifier characters. Additionally,
+you can use the `@JsName`and `@JsQualifier` annotations to map Kotlin property names to JavaScript equivalents:
```
Comment (AlejandraPedroza) [inline docs/topics/whatsnew-eap.md:356]:
```suggestion
you can use the `@JsName` and `@JsQualifier` annotations to map Kotlin property names to JavaScript equivalents:
```

File: docs/topics/whatsnew-eap.md (hunk 1 above)
Comment (koshachy) [reply docs/topics/whatsnew-eap.md:159]: It’s more of a nitpick. I’d like to keep it as it is.

File: docs/topics/whatsnew-eap.md [hunk 3]
```diff
@@ -139,6 +402,47 @@ Starting with Kotlin %kotlinEapVersion%, the minimum supported Android Gradle pl
...
+| `PLATFORM_CLASS_MAPPED_TO_KOTLIN`                     | `java.lang.String` is used instead of `kotlin.String`                                    |
+| `ARRAY_EQUALITY_OPERATOR_CAN_BE_REPLACED_WITH_EQUALS` | `arrayOf("") == arrayOf("")` is used instead of `arrayOf("").contentEquals(arrayOf(""))` |
+| `REDUNDANT_CALL_OF_CONVERSION_METHOD`                 | `42.toInt()` is used instead of `42`                                                     |
+| `USELESS_CALL_ON_NOT_NULL`                            | ` "".orEmpty()` is used instead of `""`                                                  |
+| `REDUNDANT_SINGLE_EXPRESSION_STRING_TEMPLATE`         | `"$string"` is used instead of `string`                                                  |
+| `UNUSED_ANONYMOUS_PARAMETER`                          | A parameter is passed in the lambda expression but never used                            |
+| `REDUNDANT_VISIBILITY_MODIFIER`                       | `public class Klass` is used instead of `class Klass`                                    |
+| `REDUNDANT_MODALITY_MODIFIER`                         | `final class Klass` is used instead of class Klass                                       |
```
Comment (AlejandraPedroza) [inline docs/topics/whatsnew-eap.md:421]: `class Klass`  as the previous one?

File: docs/topics/whatsnew-eap.md [hunk 4]
```diff
@@ -67,19 +67,110 @@ the Kotlin team at JetBrains is focusing on improving the language with new feat
...
 [See the full list of Kotlin language design features and proposals](kotlin-language-features-and-proposals.md).
 
+## New Gradle DSL for compiler options in multiplatform projects is stable
+
+In Kotlin 2.0.0, [we introduced a new Experimental Gradle DSL](whatsnew20.md#new-gradle-dsl-for-compiler-options-in-multiplatform-projects) 
+to simplify the configuration of compiler options across your multiplatform projects. 
+In Kotlin %kotlinEapVersion%, this DSL has been promoted to Stable.
+
+With this new DSL, you can configure compiler options at the extension level for all targets and shared source sets 
+like `commonMain`, and at the target level for specific targets:
```
Comment (AlejandraPedroza) [inline docs/topics/whatsnew-eap.md:90]:
```suggestion
(like `commonMain`), as well as at the target level for specific targets:
```

Comment (AlejandraPedroza) [reply docs/topics/whatsnew-eap.md:90]: To make it easier to read.

**Review by AlejandraPedroza — APPROVED** [review]
👍  Left minor comments

**Pull request merged**